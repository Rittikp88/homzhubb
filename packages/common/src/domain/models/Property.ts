export enum TypeOfSale {
  FIND_TENANT = 'FIND_TENANT',
  SELL_PROPERTY = 'SELL_PROPERTY',
  TENANT_FOUND = 'TENANT_FOUND',
  RENT = 'RENT',
  SALE = 'SALE',
}

export interface IPropertyTypes {
  id: string | number;
  name: string;
}

export interface IPropertyDetailsData {
  id: string | number;
  name: string;
  title?: string;
  asset_types: IPropertyTypes[];
  space_types: IPropertyTypes[];
}

export interface IRentServiceList {
  id: number;
  name: TypeOfSale;
  label: string;
  icon: string;
}
