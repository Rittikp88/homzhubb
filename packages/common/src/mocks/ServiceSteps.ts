enum ServiceStepTypes {
  LEASE_DETAILS = 'LEASE_DETAILS',
  PROPERTY_IMAGES = 'PROPERTY_IMAGES',
  PROPERTY_VERIFICATIONS = 'PROPERTY_VERIFICATIONS',
  PAYMENT_TOKEN_AMOUNT = 'PAYMENT_TOKEN_AMOUNT',
}
export const ServiceSteps = [
  {
    id: 1,
    name: ServiceStepTypes.LEASE_DETAILS,
    title: 'Enter the Lease details',
  },
  {
    id: 2,
    name: ServiceStepTypes.PROPERTY_IMAGES,
    title: 'Add Property Images',
  },
  {
    id: 3,
    name: ServiceStepTypes.PROPERTY_VERIFICATIONS,
    title: 'Complete property verification',
  },
  {
    id: 4,
    name: ServiceStepTypes.PAYMENT_TOKEN_AMOUNT,
    title: 'Payment of token amount',
  },
];
