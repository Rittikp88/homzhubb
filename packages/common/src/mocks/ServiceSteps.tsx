export const ServiceSteps = [
  {
    name: 'Free Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
    ],
  },
  {
    name: 'Assisted Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
      { id: 3, name: 'Complete property verification' },
      { id: 4, name: 'Payment of token amount' },
    ],
  },
  {
    name: 'Boosted Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
      { id: 3, name: 'Complete property verification' },
      { id: 4, name: 'Payment of token amount' },
    ],
  },
];
