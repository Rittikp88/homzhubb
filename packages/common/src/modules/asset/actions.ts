import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { IFluxStandardAction } from '@homzhub/common/src/modules/interfaces';
import { IGetAssetPayload, IGetDocumentPayload } from '@homzhub/common/src/modules/asset/interfaces';
import { Asset, IAsset } from '@homzhub/common/src/domain/models/Asset';
import { AssetDocument } from '@homzhub/common/src/domain/models/AssetDocument';
import { AssetReview, IAssetReview } from '@homzhub/common/src/domain/models/AssetReview';
import { AssetVisit, IAssetVisit } from '@homzhub/common/src/domain/models/AssetVisit';
import { IAssetVisitPayload } from '@homzhub/common/src/domain/repositories/interfaces';

const actionTypePrefix = 'Asset/';

export const AssetActionTypes = {
  GET: {
    ASSET: `${actionTypePrefix}ASSET`,
    ASSET_SUCCESS: `${actionTypePrefix}ASSET_SUCCESS`,
    ASSET_FAILURE: `${actionTypePrefix}ASSET_FAILURE`,
    REVIEWS: `${actionTypePrefix}REVIEW`,
    REVIEWS_SUCCESS: `${actionTypePrefix}REVIEW_SUCCESS`,
    REVIEWS_FAILURE: `${actionTypePrefix}REVIEW_FAILURE`,
    ASSET_DOCUMENT: `${actionTypePrefix}ASSET_DOCUMENT`,
    ASSET_DOCUMENT_SUCCESS: `${actionTypePrefix}ASSET_DOCUMENT_SUCCESS`,
    ASSET_DOCUMENT_FAILURE: `${actionTypePrefix}ASSET_DOCUMENT_FAILURE`,
    ASSET_VISIT: `${actionTypePrefix}ASSET_VISIT`,
    ASSET_VISIT_SUCCESS: `${actionTypePrefix}ASSET_VISIT_SUCCESS`,
    ASSET_VISIT_FAILURE: `${actionTypePrefix}ASSET_VISIT_FAILURE`,
  },
  SET: {
    VISIT_IDS: `${actionTypePrefix}VISIT_IDS`,
  },
  CLEAR_ASSET: `${actionTypePrefix}CLEAR_ASSET`,
};

const getAssetReviews = (id: number): IFluxStandardAction<number> => ({
  type: AssetActionTypes.GET.REVIEWS,
  payload: id,
});

const getAssetReviewsSuccess = (payload: AssetReview[]): IFluxStandardAction<IAssetReview[]> => ({
  type: AssetActionTypes.GET.REVIEWS_SUCCESS,
  payload: ObjectMapper.serializeArray(payload),
});

const getAssetReviewsFailure = (error: string): IFluxStandardAction => ({
  type: AssetActionTypes.GET.REVIEWS_FAILURE,
  error,
});

const getAsset = (payload: IGetAssetPayload): IFluxStandardAction<IGetAssetPayload> => ({
  type: AssetActionTypes.GET.ASSET,
  payload,
});

const getAssetSuccess = (asset: Asset): IFluxStandardAction<IAsset> => ({
  type: AssetActionTypes.GET.ASSET_SUCCESS,
  payload: ObjectMapper.serialize(asset),
});

const getAssetFailure = (error: string): IFluxStandardAction => ({
  type: AssetActionTypes.GET.ASSET_FAILURE,
  error,
});

const clearAsset = (): IFluxStandardAction => ({
  type: AssetActionTypes.CLEAR_ASSET,
});

const getAssetDocument = (payload: IGetDocumentPayload): IFluxStandardAction<IGetDocumentPayload> => ({
  type: AssetActionTypes.GET.ASSET_DOCUMENT,
  payload,
});

const getAssetDocumentSuccess = (data: AssetDocument[]): IFluxStandardAction<AssetDocument[]> => ({
  type: AssetActionTypes.GET.ASSET_DOCUMENT_SUCCESS,
  payload: data,
});

const getAssetDocumentFailure = (error: string): IFluxStandardAction<string> => ({
  type: AssetActionTypes.GET.ASSET_DOCUMENT_FAILURE,
  error,
});

const getAssetVisit = (payload: IAssetVisitPayload): IFluxStandardAction<IAssetVisitPayload> => ({
  type: AssetActionTypes.GET.ASSET_VISIT,
  payload,
});

const getAssetVisitSuccess = (payload: AssetVisit[]): IFluxStandardAction<IAssetVisit[]> => ({
  type: AssetActionTypes.GET.ASSET_VISIT_SUCCESS,
  payload: ObjectMapper.serializeArray(payload),
});

const getAssetVisitFailure = (error: string): IFluxStandardAction<string> => ({
  type: AssetActionTypes.GET.ASSET_VISIT_FAILURE,
  error,
});

const setVisitIds = (payload: number[]): IFluxStandardAction<number[]> => ({
  type: AssetActionTypes.SET.VISIT_IDS,
  payload,
});

export type AssetPayloadTypes = number | IAssetReview[] | IAsset | IAssetVisit[] | number[];

export const AssetActions = {
  clearAsset,
  getAssetReviews,
  getAssetReviewsSuccess,
  getAssetReviewsFailure,
  getAsset,
  getAssetSuccess,
  getAssetFailure,
  getAssetDocument,
  getAssetDocumentSuccess,
  getAssetDocumentFailure,
  getAssetVisit,
  getAssetVisitSuccess,
  getAssetVisitFailure,
  setVisitIds,
};
