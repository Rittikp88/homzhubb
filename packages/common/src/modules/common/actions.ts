import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { IRedirectionDetails } from '@homzhub/mobile/src/services/LinkingService';
import { IFluxStandardAction } from '@homzhub/common/src/modules/interfaces';
import { Country, ICountry } from '@homzhub/common/src/domain/models/Country';
import { Messages } from '@homzhub/common/src/domain/models/Message';
import { GroupMessage } from '@homzhub/common/src/domain/models/GroupMessage';
import { IGetMessageParam } from '@homzhub/common/src/domain/repositories/interfaces';

const actionTypePrefix = 'Common/';
export const CommonActionTypes = {
  GET: {
    COUNTRIES: `${actionTypePrefix}COUNTRIES`,
    COUNTRIES_SUCCESS: `${actionTypePrefix}COUNTRIES_SUCCESS`,
    MESSAGES: `${actionTypePrefix}MESSAGES`,
    MESSAGES_SUCCESS: `${actionTypePrefix}MESSAGES_SUCCESS`,
    GROUP_MESSAGES: `${actionTypePrefix}GROUP_MESSAGES`,
    GROUP_MESSAGES_SUCCESS: `${actionTypePrefix}GROUP_MESSAGES_SUCCESS`,
  },
  SET: {
    DEVICE_COUNTRY: `${actionTypePrefix}DEVICE_COUNTRY`,
    REDIRECTION_DETAILS: `${actionTypePrefix}REDIRECTION_DETAILS`,
  },
};

const getCountries = (): IFluxStandardAction => ({
  type: CommonActionTypes.GET.COUNTRIES,
});

const getCountriesSuccess = (payload: Country[]): IFluxStandardAction<ICountry[]> => ({
  type: CommonActionTypes.GET.COUNTRIES_SUCCESS,
  payload: ObjectMapper.serializeArray(payload),
});

const setDeviceCountry = (countryCode: string): IFluxStandardAction<string> => ({
  type: CommonActionTypes.SET.DEVICE_COUNTRY,
  payload: countryCode,
});

const setRedirectionDetails = (payload: IRedirectionDetails): IFluxStandardAction<IRedirectionDetails> => ({
  type: CommonActionTypes.SET.REDIRECTION_DETAILS,
  payload,
});

const getMessages = (payload: IGetMessageParam): IFluxStandardAction<IGetMessageParam> => ({
  type: CommonActionTypes.GET.MESSAGES,
  payload,
});

const getMessagesSuccess = (payload: Messages): IFluxStandardAction<Messages> => ({
  type: CommonActionTypes.GET.MESSAGES_SUCCESS,
  payload,
});

// TODO: (Shivam: 23/2/21: add types)
const getGroupMessage = (): IFluxStandardAction => ({
  type: CommonActionTypes.GET.GROUP_MESSAGES,
});

const getGroupMessageSuccess = (groupMessages: GroupMessage[]): IFluxStandardAction<GroupMessage[]> => ({
  type: CommonActionTypes.GET.GROUP_MESSAGES_SUCCESS,
  payload: ObjectMapper.serializeArray(groupMessages),
});

export type CommonActionPayloadTypes =
  | ICountry[]
  | IRedirectionDetails
  | IGetMessageParam
  | Messages
  | string
  | number
  | GroupMessage[];

export const CommonActions = {
  getCountries,
  getCountriesSuccess,
  setDeviceCountry,
  setRedirectionDetails,
  getMessages,
  getMessagesSuccess,
  getGroupMessage,
  getGroupMessageSuccess,
};
