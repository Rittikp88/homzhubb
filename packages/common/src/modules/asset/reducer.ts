import { IFluxStandardAction } from '@homzhub/common/src/modules/interfaces';
import { IAssetState } from '@homzhub/common/src/modules/asset/interfaces';
import { AssetActionTypes, AssetPayloadTypes } from '@homzhub/common/src/modules/asset/actions';
import { IAsset } from '@homzhub/common/src/domain/models/Asset';
import { IAssetReview } from '@homzhub/common/src/domain/models/AssetReview';

const initialAssetState: IAssetState = {
  error: {
    reviews: '',
    asset: '',
  },
  loaders: {
    reviews: false,
    asset: false,
  },
  asset: null,
  reviews: [],
};

export const assetReducer = (
  state: IAssetState = initialAssetState,
  action: IFluxStandardAction<AssetPayloadTypes>
): IAssetState => {
  switch (action.type) {
    case AssetActionTypes.GET.REVIEWS:
      return {
        ...state,
        ['reviews']: [],
        ['loaders']: { ...state.loaders, ['reviews']: true },
        ['error']: { ...state.error, ['reviews']: '' },
      };
    case AssetActionTypes.GET.REVIEWS_SUCCESS:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['reviews']: false },
        ['reviews']: action.payload as IAssetReview[],
      };
    case AssetActionTypes.GET.REVIEWS_FAILURE:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['reviews']: false },
        ['error']: { ...state.error, ['reviews']: action.error as string },
      };
    case AssetActionTypes.GET.ASSET:
      return {
        ...state,
        asset: null,
        ['loaders']: { ...state.loaders, ['asset']: true },
        ['error']: { ...state.error, ['asset']: '' },
      };
    case AssetActionTypes.GET.ASSET_SUCCESS:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['asset']: false },
        ['asset']: action.payload as IAsset,
      };
    case AssetActionTypes.GET.ASSET_FAILURE:
    case AssetActionTypes.POST.LEAD_FAILURE:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['asset']: false },
        ['error']: { ...state.error, ['asset']: action.error as string },
      };
    case AssetActionTypes.POST.LEAD:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['asset']: true },
        ['error']: { ...state.error, ['asset']: '' },
      };
    case AssetActionTypes.POST.LEAD_SUCCESS:
      return {
        ...state,
        ['loaders']: { ...state.loaders, ['asset']: false },
        ['error']: { ...state.error, ['asset']: '' },
      };
    case AssetActionTypes.CLEAR_ASSET:
      return initialAssetState;
    default:
      return state;
  }
};
