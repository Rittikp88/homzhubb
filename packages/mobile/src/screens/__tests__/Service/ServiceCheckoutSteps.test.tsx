// @ts-nocheck
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { initialUserState } from '@homzhub/common/src/modules/user/reducer';
import { initialPropertyState } from '@homzhub/common/src/modules/property/reducer';
import { PropertyActionTypes } from '@homzhub/common/src/modules/property/actions';
import { ServiceSteps } from '@homzhub/common/src/mocks/ServiceSteps';
import {
  ServiceCheckoutSteps,
  mapStateToProps,
  mapDispatchToProps,
} from '@homzhub/mobile/src/screens/Service/ServiceCheckoutSteps';
import { TypeOfSale } from '@homzhub/common/src/domain/models/Property';

const mock = jest.fn();

describe('Service List Steps Screen', () => {
  let component: ShallowWrapper;
  let props: any;
  let instance: any;

  beforeEach(() => {
    props = {
      steps: ServiceSteps.steps,
      propertyId: 1,
      termId: 1,
      serviceCategory: {
        id: 1,
        typeOfSale: TypeOfSale.FIND_TENANT,
      },
      navigation: {
        navigate: mock,
        goBack: mock,
      },
    };
    component = shallow(<ServiceCheckoutSteps {...props} t={(key: string): string => key} />);
    instance = component.instance();
  });

  it('should render snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot with state values', () => {
    component.setState({ currentStep: 1 });
    instance.onStepPress(1);
    instance.onProceedToNextStep();
    component.setState({ isPaymentSuccess: true });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for next steps', () => {
    component.setState({ currentStep: 5 });
    instance.onProceedToNextStep();
    instance.onSuccess();
    instance.navigateToPropertyHelper('test');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for markdown navigation', () => {
    instance.navigateToPropertyHelper('verification');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for LEASE_DETAILS title', () => {
    instance.getTitleStringsForStep('LEASE_DETAILS');
    instance.renderContent();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for LEASE_DETAILS title', () => {
    props = {
      steps: ServiceSteps.steps,
      serviceCategory: {
        id: 1,
        typeOfSale: TypeOfSale.SELL_PROPERTY,
      },
    };
    component = shallow(<ServiceCheckoutSteps {...props} t={(key: string): string => key} />);
    component.instance().getTitleStringsForStep('LEASE_DETAILS');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for PROPERTY_IMAGES title', () => {
    instance.getTitleStringsForStep('PROPERTY_IMAGES');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for PROPERTY_VERIFICATIONS title', () => {
    instance.getTitleStringsForStep('PROPERTY_VERIFICATIONS');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for PAYMENT_TOKEN_AMOUNT title', () => {
    instance.getTitleStringsForStep('PAYMENT_TOKEN_AMOUNT');
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should navigate back', () => {
    component.find('[testID="lblNavigate"]').prop('onIconPress')();
    expect(mock).toHaveBeenCalled();
  });

  it('should return service steps', () => {
    const mockedState = {
      user: {
        ...initialUserState,
      },
      property: {
        ...initialPropertyState,
        termId: 1,
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.termId).toBe(1);
  });

  it('should dispatch getServiceStepsDetails', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setTermId(1);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: PropertyActionTypes.SET.TERM_ID,
      payload: 1,
    });
  });
});
