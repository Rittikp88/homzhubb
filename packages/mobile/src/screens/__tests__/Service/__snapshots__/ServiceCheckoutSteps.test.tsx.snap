// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service List Steps Screen should render snapshot 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for LEASE_DETAILS title 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for LEASE_DETAILS title 2`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for PAYMENT_TOKEN_AMOUNT title 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for PROPERTY_IMAGES title 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for PROPERTY_VERIFICATIONS title 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for markdown navigation 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={0}
    isPaymentSuccess={false}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot for next steps 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={5}
    isPaymentSuccess={true}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
  <WithShadowView
    outerViewStyle={
      Object {
        "marginBottom": 20,
        "paddingBottom": 0,
        "paddingTop": 10,
      }
    }
  >
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "margin": 16,
        }
      }
      title="previewProperty"
      type="primary"
    />
  </WithShadowView>
</Fragment>
`;

exports[`Service List Steps Screen should render snapshot with state values 1`] = `
<Fragment>
  <Header
    backgroundColor="#0084F8"
    icon="left-arrow"
    iconColor="#FFFFFF"
    isHeadingVisible={true}
    onIconPress={[Function]}
    testID="lblNavigate"
    title=""
    titleFontType="semiBold"
    titleType="small"
  />
  <StepIndicatorComponent
    containerStyle={
      Object {
        "backgroundColor": "#0084F8",
        "paddingVertical": 12,
      }
    }
    currentPosition={2}
    isPaymentSuccess={true}
    labels={
      Array [
        "",
        "",
        "",
        "",
      ]
    }
    onPress={[Function]}
    stepCount={4}
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginTop": 20,
        }
      }
    >
      <Label
        style={
          Object {
            "color": "#33495E",
          }
        }
        textType="semiBold"
        type="large"
      >
        step
      </Label>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#0084F8",
          }
        }
        textType="semiBold"
        type="small"
      >
        skipForNow
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="regular"
    />
  </ScrollViewMock>
  <WithShadowView
    outerViewStyle={
      Object {
        "marginBottom": 20,
        "paddingBottom": 0,
        "paddingTop": 10,
      }
    }
  >
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "margin": 16,
        }
      }
      title="previewProperty"
      type="primary"
    />
  </WithShadowView>
</Fragment>
`;
